test/NuGet.Clients.Tests/NuGet.PackageManagement.UI.Test/Models/PackageItemListViewModelTests.cs

```diff
@@ -37,5 +54,424 @@ public void LocalSources_PackageReader_NotNull()
+                _output.WriteLine($"Url {builder.Uri}");
+
+                // Act
+                BitmapSource result = await GetFinalIconBitmapAsync(packageItemListViewModel);
+
+                VerifyImageResult(result, packageItemListViewModel.BitmapStatus);
+
+                _output.WriteLine($"result {result}");
+                string resultFormat = result != null ? result.Format.ToString() : "";
+                _output.WriteLine($"Pixel format: {resultFormat}");
+
+                // Assert
+                Assert.Equal(IconBitmapStatus.DefaultIconDueToDecodingError, packageItemListViewModel.BitmapStatus);
+            }
+        }
+
+        private static void VerifyImageResult(object result, IconBitmapStatus bitmapStatus)
+        {
+            if (result == null && IconBitmapStatusUtility.GetIsDefaultIcon(bitmapStatus))
+            {
+                return;
+            }
+            else
+            {
+                Assert.NotNull(result);
+                Assert.True(result is BitmapImage || result is CachedBitmap);
+                var image = result as BitmapSource;
+                Assert.NotNull(image);
+                Assert.Equal(32, image.PixelWidth);
+                Assert.Equal(32, image.PixelHeight);
+            }
+        }
+
+        /// <summary>
+        /// Creates a NuGet package with .nupkg extension and with a PNG image named "icon.png"
+        /// </summary>
+        /// <param name="path">Path to NuGet package</param>
+        /// <param name="iconName">Icon filename with .png extension</param>
+        private static void CreateDummyPackage(
+            string zipPath,
+            string iconName = "icon.png",
+            string iconFile = "icon.png",
+            string iconFileSourceElement = "icon.png",
+            string iconFileTargetElement = "",
+            bool isRealImage = true)
+        {
+            var dir = Path.GetDirectoryName(zipPath);
+            var holdDir = "pkg";
+            var folderPath = Path.Combine(dir, holdDir);
+
+            // base dir
+            Directory.CreateDirectory(folderPath);
+
+            // create nuspec
+            var nuspec = NuspecBuilder.Create()
+                .WithIcon(iconName)
+                .WithFile(iconFileSourceElement, iconFileTargetElement);
+
+            // create png image
+            var iconPath = Path.Combine(folderPath, iconFile);
+            var iconDir = Path.GetDirectoryName(iconPath);
+            Directory.CreateDirectory(iconDir);
+
+            if (isRealImage)
+            {
+                CreateNoisePngImage(iconPath);
+            }
+            else
+            {
+                File.WriteAllText(iconPath, "I am an image");
+            }
+
+            // Create nuget package
+            using (var nuspecStream = new MemoryStream())
+            using (FileStream nupkgStream = File.Create(zipPath))
+            {
+                var writer = new StreamWriter(nuspecStream);
```

Comment:
Wrap with `using` statement.

Reply:
my first triple using!!!  fixed in next iteration.